### ISSUE TITLE
Feature Request - Keyboard support for Step element

### ISSUE THREAD

**Issue by Slapbox** [reakit/reakit#217]
It would be great to be able to optionally use the left and right arrow keys to move through steps.

**Comment by renatorib** [comment]
I'm studying how I can do this. I noticed that Tabs have support to left/right when TabsTab are focused. It makes sense for tabs. But steps has a more generic proposal, it does not necessarily have pointers (next/prev) because it is usually used as low level to others components that implement your navigation rules (TabsTab is an example of implementation using `show` method), where and how would be the best place to apply in this case? 

@diegohaz @Thomazella do you think it's worth investing time in this feature?

**Comment by diegohaz** [comment]
I agree that `Step` may be the wrong place to put this keyboard feature. Maybe it should be abstracted into its own component, just like we did with `Portal` (which was inside `Overlay` before) in #169.

But I'd like to see more use cases before doing it (and possibly doing it the wrong way). #242 may be one.

I think it's worth the time as more use cases appear. Most of them should probably be resolved using `Tabs` though (I've already made a Carousel from it, for example).
